src/common/exception/GraknClientException.jl

```diff
@@ -2,13 +2,35 @@
 
 mutable struct GraknClientException <: Exception
     error_message::T where {T<:Union{<:AbstractGeneralError,<:Nothing}}
-    params
+    params::Optional{Tuple}
     individual_message::Union{Nothing, String}
-    cause::R where {R<:Exception}
+    cause::R where {R<:Union{Nothing, Exception}}
+end
+
+Base.show(io::IO, grakn_excption::GraknClientException) = Base.print(io,grakn_excption)
+function Base.print(io::IO, grakn_excption::GraknClientException)
```

Comment:
The general technique is to put your logic in `Base.show(io::IO, whatever)`. Extending `Base.print` may have undesirable consequence... https://docs.julialang.org/en/v1/manual/types/#man-custom-pretty-printing

Same comment for other code in this PR.

Reply:
Ok. I will change this across the source.